frontend/src/api/task.ts

```diff
@@ -0,0 +1,68 @@
+import { post } from "src/api/requests";
+
+/**
+ * Initializes the shape for the data being retrieved from the backend.
+ */
+export interface Task {
+  _id: string;
+  title: string;
+  description?: string;
+  isChecked: boolean;
+  dateCreated: Date;
+}
+
+interface TaskJSON {
```

Comment:
```suggestion
/**
 * Defines the shape of JSON that we'll receive from the backend when we ask the API
 * for a Task object. That is, when the backend sends us a JSON object representing a
 * Task, we expect it to match these fields and types.
 *
 * The difference between this type and `Task` above is that `dateCreated` is a string
 * instead of a Date object. This is because JSON doesn't support Dates, so we use a
 * date-formatted string in requests and responses.
 */
interface TaskJSON {
```